auth-mailchimp-sync/functions/src/index.ts

```diff
@@ -95,5 +99,83 @@ export const removeUserFromList = functions.handler.auth.user.onDelete(
     } catch (err) {
       logs.errorRemoveUser(err);
     }
-  }
-);
+  });
+
+export const addExistingUsersToList = functions.tasks
+  .taskQueue()
+  .onDispatch(async (data) => {
+    const runtime = getExtensions().runtime();
+    if (!config.doBackfill) {
+      await runtime.setProcessingState(
+        "PROCESSING_COMPLETE",
+        "No existing documents imported into BigQuery."
+      );
+      return;
+    }
+    const nextPageToken = data.nextPageToken;
+    const pastSuccessCount = (data["successCount"] as number) ?? 0;
+    const pastErrorCount = (data["errorCount"] as number) ?? 0;
+
+    const res = await getAuth().listUsers(100, nextPageToken);
+    const mailchimpPromises = res.users.map(async (user) => {
+      const { email, uid } = user;
+      if (!email) {
+        logs.userNoEmail();
+        return;
+      }
+
+      try {
+        logs.userAdding(uid, config.mailchimpAudienceId);
+        const results = await mailchimp.post(
+          `/lists/${config.mailchimpAudienceId}/members`,
+          {
+            email_address: email,
+            status: config.mailchimpContactStatus,
+          }
+        );
+        logs.userAdded(
+          uid,
+          config.mailchimpAudienceId,
+          results.id,
+          config.mailchimpContactStatus
+        );
+      } catch (err) {
+        logs.errorAddUser(err);
+      }
+    });
+    const results = await Promise.allSettled(mailchimpPromises);
+    const newSucessCount =
+      pastSuccessCount + results.filter((p) => p.status === "fulfilled").length;
+    const newErrorCount =
+      pastErrorCount + results.filter((p) => p.status === "rejected").length;
+    if (res.pageToken) {
+      const queue = getFunctions().taskQueue(
+        "fsimportexistingdocs",
+        process.env.EXT_INSTANCE_ID
+      );
+      await queue.enqueue({
+        nextPageToken: res.pageToken,
+        successCount: newSucessCount,
+        errorCount: newErrorCount,
+      });
+    } else {
+      logs.backfillComplete(newSucessCount, newErrorCount);
```

Comment:
Since this is a sample, we probably want to add a comment here that says something like "The backfill is now complete, set the processing status" or something like that to make it clear that this only happens at the end of the whole backfill, and not after every page. etc.